From cccf66de67861480490d5c553516a9556f1efffc Mon Sep 17 00:00:00 2001
From: Andy Sun <bins@codeaurora.org>
Date: Fri, 28 Oct 2016 15:15:53 +0800
Subject: [PATCH 1/1] QCamera2: Add support for 64bit/32bit co-existence

This change will make a few camera structure size same
on both 64bit and 32bit platform. This will enable 64bit
processes to interface with 32bit camera daemon for proper
socket communication and memory mapping.

Change-Id: I92fcee95f34c71ce3bd5101c7583f167c21c1512
Signed-off-by: Andy Sun <bins@codeaurora.org>
---
 QCamera2/stack/common/cam_intf.h  |  2 +-
 QCamera2/stack/common/cam_types.h | 18 +++++++++---------
 2 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/QCamera2/stack/common/cam_intf.h b/QCamera2/stack/common/cam_intf.h
index 31e6246..fde1819 100644
--- a/QCamera2/stack/common/cam_intf.h
+++ b/QCamera2/stack/common/cam_intf.h
@@ -185,7 +185,7 @@ typedef struct{
     cam_format_t supported_raw_fmts[CAM_FORMAT_MAX];
 
     cam_channel_info_t supported_channel_info[MAX_CID];
-    size_t supported_channel_info_cnt;
+    uint32_t supported_channel_info_cnt;
 
     /* The minimum frame duration that is supported for above
        raw resolution */
diff --git a/QCamera2/stack/common/cam_types.h b/QCamera2/stack/common/cam_types.h
index 975dcb3..ba903ff 100644
--- a/QCamera2/stack/common/cam_types.h
+++ b/QCamera2/stack/common/cam_types.h
@@ -1407,19 +1407,19 @@ typedef struct {
 
 typedef struct {
     uint32_t tuning_data_version;
-    size_t tuning_sensor_data_size;
-    size_t tuning_vfe_data_size;
-    size_t tuning_cpp_data_size;
-    size_t tuning_cac_data_size;
-    size_t tuning_cac_data_size2;
-    size_t tuning_mod3_data_size;
+    uint32_t tuning_sensor_data_size;
+    uint32_t tuning_vfe_data_size;
+    uint32_t tuning_cpp_data_size;
+    uint32_t tuning_cac_data_size;
+    uint32_t tuning_cac_data_size2;
+    uint32_t tuning_mod3_data_size;
     uint8_t  data[TUNING_DATA_MAX];
 }tuning_params_t;
 
 typedef struct {
     int32_t event_type;
     cam_dimension_t dim;
-    size_t size;
+    uint32_t size;
     char path[QCAMERA_MAX_FILEPATH_LENGTH];
     cam_format_t picture_format;
 } cam_int_evt_params_t;
@@ -2106,12 +2106,12 @@ typedef struct {
 } cam_tonemap_curve_t;
 
 typedef struct {
-   size_t tonemap_points_cnt;
+   uint32_t tonemap_points_cnt;
    cam_tonemap_curve_t curves[3];
 } cam_rgb_tonemap_curves;
 
 typedef struct {
-   size_t tonemap_points_cnt;
+   uint32_t tonemap_points_cnt;
    cam_tonemap_curve_t curve;
 } cam_profile_tone_curve;
 
-- 
1.9.1

